src/sources/soundsourceproxy.cpp

```diff
@@ -165,6 +168,16 @@ void registerReferenceSoundSourceProviders(
 #endif
 }
 
+QList<QMimeType> mimeTypesForFileType(const QString& fileType) {
+    if (fileType == "opus") {
+        // opus has mime type "audio/ogg" which will be decoded with the SoundSourceOggVobis()
+        // but we want SoundSourceOpus()
+        return {};
```

Comment:
this seems a little hacky, is there a way we can detect files as audio/ogg and then later decide if they are vorbis or opus? It's not a bug that opus shows up as an ogg mimetype, it's just that mimetype is not sufficient to determine the codec, much like .avi can contain many codecs. (https://datatracker.ietf.org/doc/html/rfc7845)

Reply:
We may create a joint sound source for opus and vorbis files. Here we just work around the limitation of  QMimeDatabase() because we currently need both, the mime type + the extension to select the soundsource.